js/models/Submission.js

```diff
@@ -4,9 +4,11 @@ export default class Submission {
 Submission.schema = {
   name: 'Submission',
   properties: {
+    uniqueId: 'string',
```

Comment:
Adding `primaryKey: 'uniqueId'` to the schema can enforce uniqueness and make lookups more efficient.